engine/.rubocop.yml

```diff
@@ -6,15 +6,20 @@ inherit_gem:
 AllCops:
   SuggestExtensions: false
   # Match minimum target version to gemspec
-  TargetRubyVersion: 3.0
+  TargetRubyVersion: 3.1
 
-# Calls to super in subclasses of ViewComponent::Base aren't needed
-# by default and can change the behaviour of **kwargs so we
-# exclude component classes from this rule
-# https://github.com/ViewComponent/view_component/discussions/1545
 Lint/MissingSuper:
   Exclude:
+    # Calls to super in subclasses of ViewComponent::Base aren't needed
+    # by default and can change the behaviour of **kwargs so we
+    # exclude component classes from this rule
+    # https://github.com/ViewComponent/view_component/discussions/1545
     - 'app/components/**/*'
 
+Style/HashSyntax:
+  # We do not necessarily find that Ruby 3.1’s hash value shorthand syntax
+  # encourages greater readability so support either style
+  EnforcedShorthandSyntax: 'either'
```

Comment:
In our products we've often disabled this rule. This felt like the most flexible option.